client/src/components/pages/HomePage/AclSearchResult.tsx

```diff
@@ -0,0 +1,289 @@
+import React, { useState, ReactNode, useRef } from 'react';
+import styled, { css } from 'styled-components';
+import { ChevronsDown, Link as LinkIcon } from 'react-feather';
+import Highlighter from 'react-highlight-words';
+import { Link } from 'react-router-dom';
+
+import { LinkStyle, BodySmall, FadeInText } from '../../../shared/Styles';
+import {
+  API_BASE,
+  SEARCH_COLLAPSED_ENDPOINT,
+  SEARCH_EXPANDED_ENDPOINT,
+  SEARCH_CLICKED_ENDPOINT,
+  RELATED_ROUTE,
+} from '../../../shared/Constants';
+import { makePOSTRequest, parseAbstract } from '../../../shared/Util';
+import { SearchArticle } from '../../../shared/Models';
+import AclBaseArticleResult from '../../common/AclBaseArticleResult';
+
+interface SearchResultProps {
+  article: SearchArticle;
+  position: number;
+  queryId: string;
+  queryTokens: Array<string>;
+}
+
+const highlightText = (
+  text: string,
+  highlights: Array<[number, number]>,
+  queryTokens: Array<string>,
+): Array<string | ReactNode> => {
+  if (!highlights) {
+    return [
+      <TextSpan className="hideCollapsed" key={0}>
+        {text}
+      </TextSpan>,
+    ];
+  }
+
+  let highlighted: Array<string | ReactNode> = [];
+  let prevEnd = -1;
+
+  highlights.forEach((highlight, i) => {
+    const [start, end] = highlight;
+    highlighted.push(
+      <Ellipsis key={`${i}-ellipsis-1`} className="showCollapsed">
+        ...
+      </Ellipsis>,
+    );
+    highlighted.push(
+      <TextSpan key={`${i}-1`} className="hideCollapsed">
+        {highlightMatches(queryTokens, text.substr(prevEnd + 1, start - prevEnd - 1))}
+      </TextSpan>,
+    );
+    highlighted.push(
+      <Highlight className="highlight" key={`${i}-2`}>
+        {highlightMatches(queryTokens, text.substr(start, end - start + 1))}
+      </Highlight>,
+    );
+
+    prevEnd = end;
+  });
+
+  // add last part of text
+  if (prevEnd < text.length) {
+    highlighted.push(
+      <TextSpan key="last" className="hideCollapsed">
+        {highlightMatches(queryTokens, text.substr(prevEnd + 1, text.length - prevEnd - 1))}
+      </TextSpan>,
+    );
+  }
+
+  return highlighted;
+};
+
+const highlightMatches = (queryTokens: Array<string>, text: string): ReactNode => {
+  return (
+    <Highlighter
+      searchWords={queryTokens}
+      textToHighlight={text}
+      highlightTag={Match}
+      highlightClassName="match"
+    />
+  );
+};
+
+// adjust highlights based on difference
+const adjustHighlights = (
```

Comment:
can you move all these duplicated highlight functions into a `utils/highlight.tsx` file and import them?